library/dnet.c

```diff
@@ -783,6 +785,25 @@ int dnet_server_send_write(struct dnet_server_send_ctl *send,
 			(unsigned long long)re->iterated_keys, (unsigned long long)re->total_keys,
 			atomic_read(&send->bytes_pending), send->bytes_pending_max);
 
+	// check that source and destination backends are not the same
+	const int group_id = dnet_group_id_search_by_backend(n->st, send->backend_id);
+	if (group_id >= 0) {
+		for (i = 0; i < send->group_num; ++i) {
+			if (send->groups[i] == group_id) {
+				ctl.id.group_id = group_id;
+				const struct dnet_net_state *st = dnet_state_get_first_with_backend(n, &ctl.id, &backend_id);
+				if (st == n->st && backend_id == send->backend_id) {
+					dnet_log(n, DNET_LOG_ERROR, "%s: Interrupting iterator: source and destination backends are the same: "
```

Comment:
It shouldn't interrupt iteration here and there are few reasons for that:
* it should avoid writing keys to the same backend, but it should write all keys to other groups
* if there is only one destination group, it should skip keys that should be written to the same backend, but it should write all keys to other backends